### PR TITLE
add all event attribute types

### PR DESCRIPTION
**Pull request by HashZhang**
add all event attribute types

### CODE REVIEW

File: src/main/java/dev/morling/jfrunit/ExpectedEvent.java [hunk 1]
```diff
@@ -73,10 +75,37 @@ private static boolean hasMatchingProperty(RecordedEvent recordedEvent, String n
...
+            return Objects.equals(recordedEvent.getDuration(name), value);
         }
         else if (value instanceof String) {
-            return recordedEvent.getString(name).equals(value);
+            return Objects.equals(recordedEvent.getString(name), value);
+        }
+        else if (value instanceof Boolean) {
+            return Objects.equals(recordedEvent.getBoolean(name), value);
+        }
+        else if (value instanceof Boolean) {
```
Comment (gunnarmorling) [inline src/main/java/dev/morling/jfrunit/ExpectedEvent.java:86]:
```suggestion
        else if (value instanceof Byte) {
```

File: src/main/java/dev/morling/jfrunit/ExpectedEvent.java [hunk 2]
```diff
@@ -73,10 +75,37 @@ private static boolean hasMatchingProperty(RecordedEvent recordedEvent, String n
         // TODO all event attribute types
```
Comment (gunnarmorling) [inline src/main/java/dev/morling/jfrunit/ExpectedEvent.java:75]: TODO comment is obsolete now.

**Review by gunnarmorling — CHANGES_REQUESTED** [review]
Thanks a lot! Two small comments inline. Can the value also be of a primitive type (e.g. `int` instead of `Integer`)? If so, this would need consideration too. And could you add some test too? Probably the easiest would be a custom event with one attribute of each type.

Lastly, could you author the commit with an email address associated to your GitHub account?

Thanks a lot!

**Comment by HashZhang** [comment]
ok, thanks, will be done soon

File: src/main/java/dev/morling/jfrunit/ExpectedEvent.java [hunk 3]
```diff
@@ -70,13 +72,38 @@ private static boolean hasMatchingProperty(RecordedEvent recordedEvent, String n
...
-
         if (value instanceof Duration) {
-            return recordedEvent.getDuration(name).equals(value);
+            return Objects.equals(recordedEvent.getDuration(name), value);
         }
         else if (value instanceof String) {
-            return recordedEvent.getString(name).equals(value);
+            return Objects.equals(recordedEvent.getString(name), value);
+        }
+        else if (value instanceof Byte) {
```
Comment (ebramirez) [inline src/main/java/dev/morling/jfrunit/ExpectedEvent.java:81]:
```suggestion
        else if (value instanceof Boolean) {
```

**Comment by HashZhang** [comment]
I will open another pull request to add more rigorous logic.

**Pull request closed without merging**